packages/common/auth-token.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
```

Comment:
This is a possible candidate for class objects given the config mapping.   That said, the implementation would be similar with the constructors handling it and having it remain purely functional felt preferable.